src/Lookup/MultiplexingEntityLookup.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityDocument;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikimedia\Assert\Assert;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterElementTypeException;
+
+/**
+ * Wrapper around repository-specific EntityLookups
+ * picking up the right lookup object for the particular input.
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class MultiplexingEntityLookup implements EntityLookup {
```

Comment:
Renamed to `DispatchingEntityLookup` for now. Could come up with a nice way to include the repository name in the name (`RepositoryNameBasedDispatchingEntityLookup` is not the nicest name) but any suggestions are welcome!